compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -3119,7 +3119,8 @@ define_print! {
 
     ty::ExistentialProjection<'tcx> {
         let name = cx.tcx().associated_item(self.def_id).name;
-        p!(write("{} = ", name), print(self.term))
+        let args = &self.args[cx.tcx().generics_of(self.def_id).parent_count - 1..];
```

Comment:
If you feel uneasy about the `- 1` (which won't ever panic as traits always have a SelfTyParam) I can switch to converting the existential projection into a normal projection predicate with `with_self_ty(trait_object_dummy_self)` and replace the explicit slicing with `ty::Generics::own_args`. That's heavier tho perf-wise I presume.

Reply:
It's whatever. Please leave a comment explaining what the `-1` is doing though.